src/main/java/io/cdap/plugin/gcp/firestore/util/FirestoreUtil.java

```diff
@@ -38,31 +38,82 @@ public class FirestoreUtil {
    *
-   * @param serviceAccountFilePath path to credentials defined in JSON file
+   * @param serviceAccount Actual credentials in JSON or a path to the file containing them
+   * @param isServiceAccountFilePath If the serviceAccount parameter is a file path or not
    * @param projectId Google Cloud project ID
+   * @param databaseId Cloud Firestore Database Id
    * @return Firestore service
    */
-  public static Firestore getFirestore(@Nullable String serviceAccountFilePath, String projectId) {
+  public static Firestore getFirestore(@Nullable String serviceAccount,
+                                       @Nullable Boolean isServiceAccountFilePath,
+                                       String projectId,
+                                       String databaseId) {
     try {
       FirestoreOptions.Builder optionsBuilder = FirestoreOptions.newBuilder()
-        .setProjectId(projectId);
-
-      LOG.debug("serviceAccount={}, project={}...", serviceAccountFilePath, projectId);
-
-      if (!Strings.isNullOrEmpty(serviceAccountFilePath)) {
-        optionsBuilder.setCredentials(loadServiceAccountCredentials(serviceAccountFilePath));
-      }
-
+        .setProjectId(projectId)
+        .setDatabaseId(databaseId);
+      LOG.debug("isServiceAccountFilePath={}, project={}, databaseId={}...",
+        isServiceAccountFilePath, projectId, databaseId);
+      final GoogleCredentials credential = getCredential(serviceAccount, isServiceAccountFilePath);
+      optionsBuilder.setCredentials(credential);
       return optionsBuilder.build().getService();
     } catch (IOException e) {
       throw new FirestoreInitializationException("Unable to connect to Firestore", e);
     }
   }
 
-  public static ServiceAccountCredentials loadServiceAccountCredentials(String path) throws IOException {
+  public static GoogleCredentials loadServiceAccountCredentials(String path) throws IOException {
     File credentialsPath = new File(path);
     try (FileInputStream serviceAccountStream = new FileInputStream(credentialsPath)) {
-      return ServiceAccountCredentials.fromStream(serviceAccountStream);
+      return GoogleCredentials.fromStream(serviceAccountStream);
+    }
+  }
+
+  /**
+   * Obtains Google Cloud Service Account credential from given JSON file.
+   * If give path is null or empty, obtains application default credentials.
+   *
+   * @param serviceAccount path to credentials defined in JSON file
+   * @param isServiceAccountFilePath indicator whether service account is file path or JSON
+   * @return Google Cloud Service Cloud credential
+   * @throws IOException if the credential cannot be created in the current environment
+   */
+  private static GoogleCredentials getCredential(@Nullable String serviceAccount,
+                                                @Nullable Boolean isServiceAccountFilePath) throws IOException {
+                                                GoogleCredentials credential;
```

Comment:
`GoogleCredentials credential` should be aligned with `i` in `private` on line no 81.